lib/arithmetic.js

```diff
@@ -159,12 +228,22 @@ exports.traceExpression = function(builder, expression, traceOpts) {
   // expand any globbing operations in the expression
   var expandPromise = Promise.resolve();
   operations.forEach(function(operation) {
-    expandPromise = expandPromise.then(function() {
-      return Promise.resolve(expandGlobAndCanonicalize(builder, operation))
-      .then(function(expanded) {
-        expandedOperations = expandedOperations.concat(expanded);
+    if (operation.bulkOperation) {
+      var expandedTreePromise = traceExpression(builder, operation.bulkOperation, traceOpts)
+      expandPromise = expandPromise.then(function() {
+        return Promise.resolve(expandedTreePromise)
+               .then(function(fullTree){
+                 expandedOperations = expandedOperations.concat({ preExpandedTree: fullTree, operator: operation.operator })
```

Comment:
Oh right I see what you are doing here, running the actual arithmetic during this phase recursively.

To avoid conflating steps I think I'd rather take one of two approaches here:

1. Instead of a single `traceExpression` separate it out into `traceExpression` and `traceBulkExpression`, where `traceBulkExpression` handles this sort of logic instead.
2. During this expansion step, recursively expand globs only, leaving the arithmetic syntax in-tact.

I'd be happy with either method here, let me know if that makes sense.